Mono.Debugging.Soft/SoftDebuggerAdaptor.cs

```diff
@@ -2199,113 +2205,60 @@ public MethodCall (SoftEvaluationContext ctx, MethodMirror function, object obj,
 			}
 		}
 
-		public override void Invoke ()
+		protected override void AfterCancelledImpl (int elapsedAfterCancelMs)
 		{
-			try {
-				var invocableMirror = obj as IInvocableMethodOwnerMirror;
-				if (invocableMirror != null) {
-					var optionsToInvoke = options;
-					if (obj is StructMirror) {
-						optionsToInvoke |= InvokeOptions.ReturnOutThis;
-					}
-					handle = invocableMirror.BeginInvokeMethod (ctx.Thread, function, args, optionsToInvoke, null, null);
-				}
-				else
-					throw new ArgumentException ("Soft debugger method calls cannot be invoked on objects of type " + obj.GetType ().Name);
-			} catch (InvocationException ex) {
-				ctx.Session.StackVersion++;
-				exception = ex;
-			} catch (Exception ex) {
-				ctx.Session.StackVersion++;
-				DebuggerLoggingService.LogError ("Error in soft debugger method call thread on " + GetInfo (), ex);
-				exception = ex;
-			}
 		}
 
-		public override void Abort ()
-		{
-			if (handle is IInvokeAsyncResult) {
-				var info = GetInfo ();
-				DebuggerLoggingService.LogMessage ("Aborting invocation of " + info);
-				((IInvokeAsyncResult) handle).Abort ();
-				// Don't wait for the abort to finish. The engine will do it.
-			} else {
-				throw new NotSupportedException ();
-			}
-		}
-		
-		public override void Shutdown ()
-		{
-			shutdownEvent.Set ();
-		}
-		
-		void EndInvoke ()
+		protected override Task<OperationResult<Value>> InvokeAsyncImpl (CancellationToken token)
 		{
 			try {
-				result = ((IInvocableMethodOwnerMirror) obj).EndInvokeMethodWithResult (handle);
-			} catch (InvocationException ex) {
-				if (!Aborting && ex.Exception != null) {
-					string ename = ctx.Adapter.GetValueTypeName (ctx, ex.Exception);
-					var vref = ctx.Adapter.GetMember (ctx, null, ex.Exception, "Message");
```

Comment:
I've replaced simple Exception message to full exception object. So now you can walk through it, but before you can see only exception message